File: backend/OperationHistory/OperationHistory.BLL/DataTransferObjects/SearchOperationUserDto.cs
using Common.DataTransfer;
using Common.Enum;

namespace OperationHistory.BLL.DataTransferObjects;

public class SearchOperationUserDto : PaginationRequest
{
    public List<Guid> AccountIds { get; set; } = [];
    public List<Guid> LoanIds { get; set; } = [];
    public List<CurrencyType> CurrencyTypes { get; set; } = [];
    public List<OperationType> OperationTypes { get; set; } = [];
    public List<OperationStatus> OperationStatuses { get; set; } = [];
}

File: backend-frontend/bff-client/src/dto/OperationHistory/req/SearchOperationUserDto.ts
import { CurrencyType } from 'entities/Currency'
import { OperationStatus, OperationType } from 'entities/Operation'

export type SearchOperationUserDto = {
  accountIds?: string[]
  loanIds?: string[]
  CurrencyTypes?: CurrencyType[]
  OperationTypes?: OperationType[]
  OperationStatuses?: OperationStatus[]
}
